ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -233,6 +281,25 @@ let all (lookup: string -> string option) (list: string -> string list) ~__conte
 	    Db.VM_guest_metrics.set_other ~__context ~self:gm ~value:other;
 	    Helpers.call_api_functions ~__context (fun rpc session_id -> Client.Client.VM.update_allowed_operations rpc session_id self);
 	  end;
+	  if(vif_state_cached <> vif_state) then begin
```

Comment:
Is there any reason not to just cache `network_paths_optimized` instead of building up `vif_state` and caching that?

Same for VBDs.